server/services/locationService.ts

```diff
@@ -16,8 +16,12 @@ export default class LocationService {
     )
   }
 
-  async getLocation(token: string, livingUnitId: number): Promise<Location> {
-    return await this.prisonApiClient.getLocation(token, livingUnitId)
+  async getLocation(token: string, key: string): Promise<Location> {
+    return await this.locationsInsidePrisonApiClient.getLocation(token, key)
+  }
+
+  async getInmatesAtLocation(token: string, locationId: string): Promise<Occupant[]> {
+    return await this.locationsInsidePrisonApiClient.getInmatesAtLocation(token, locationId)
```

Comment:
I feel like the `getInmatesAtLocation` method still belongs in the `prisonerCellAllocationService`, as it's to do with the prisoners at the locations, rather than the details of the location itself. I know we are getting the info from the same API as `getLocation` under the hood, so it is correct that `getInmatesAtLocation` and `getLocation` both live in the same _client_ class within the _data_ folder, as those clients map directly to external APIs, i.e. in this case both come from the locations API. But the _services_ in the _services_ folder on the other hand do not necessarily have a one-to-one relationship specific clients (if they did then they would be redundant) - they are just a way of organising the different actions logically within the project. Currently the calls to do with prisoners within locations live in one service, and the calls to do with the locations themselves live within another service. So I suggest that we either keep following that convention, or just merge those services into one.

Reply:
You are right the prisoner in the cell comes from the prisoner search service, and the locations API just wraps that, arguably the UI could simply call that API directly (like it does elsewhere that you did @jimbali ) - I added those connivence methods to the location API to hide the complexity of how to get that data.  I suggest we combine that call into one as basically its getting "location + prisoner in the location" information for the same cell.  However I think we could merge this first and add an extra parameter to the Location API to get the prisoners in the cell at the same time and then merge the two calls.  In addition there is now an endpoint that gets all cells with capacity in the locations API which means we can move off whereabouts API and prison API for this information

Reply:
ok but that's not really what I meant. I just meant that the service classes in this project each have a distinct purpose and moving the `getInmatesAtLocation` method from the `PrisonerCellAllocationService` to the `LocationService` goes against that. It should either be moved back to the `PrisonerCellAllocationService` (but it can still be updated to call our new locations API via the `locationsInsidePrisonApiClient`), or we should delete the `PrisonerCellAllocationService` class and move all of its methods to the `LocationService`. Otherwise the services won't make much sense anymore as their purposes will be muddled.

Reply:
I'm just thinking in terms of OOP principles. I see the service classes as an abstraction layer. Changing their implementation (i.e. what clients they call) shouldn't necessarily change their signature. If it does then maybe it's a code smell.

Reply:
Ok got it! @GurnankCheema - suggest we refactor after this merge

Reply:
Thank you both for reviewing
@Mjwillis do you mean refactor to move the call back into the prisonerCellAllocationService as Jim suggested or do you mean _combine that call into one as basically its getting "location + prisoner in the location" information for the same cell_ ?